ml/src/main/scala/ai/h2o/sparkling/ml/features/H2OWord2Vec.scala

```diff
@@ -0,0 +1,85 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.ml.features
+
+import ai.h2o.sparkling.H2OContext
+import ai.h2o.sparkling.ml.internals.H2OModel
+import ai.h2o.sparkling.ml.models.{H2OMOJOModel, H2OMOJOSettings}
+import ai.h2o.sparkling.ml.params.H2OWord2VecParams
+import ai.h2o.sparkling.ml.utils.EstimatorCommonUtils
+import org.apache.spark.annotation.DeveloperApi
+import org.apache.spark.ml.Estimator
+import org.apache.spark.ml.param.{Param, ParamMap}
+import org.apache.spark.ml.util.{DefaultParamsReadable, DefaultParamsWritable, Identifiable}
+import org.apache.spark.sql.Dataset
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.functions.{col, explode, udf}
+import org.apache.spark.sql.types.StructType
+
+class H2OWord2Vec(override val uid: String)
+  extends Estimator[H2OMOJOModel]
+  with H2OWord2VecParams
+  with EstimatorCommonUtils
+  with DefaultParamsWritable {
+
+  final val inputCol: Param[String] = new Param[String](this, "inputCol", "input column name")
+
+  final def getInputCol: String = $(inputCol)
+
+  final val outputCol: Param[String] = new Param[String](this, "outputCol", "output column name")
+
+  setDefault(outputCol, "H2OWord2Vec_output")
+
+  /** @group getParam */
+  final def getOutputCol: String = $(outputCol)
+
+  def this() = this(Identifiable.randomUID(classOf[H2OWord2Vec].getSimpleName))
+
+  private def addValue: UserDefinedFunction = udf((array: Seq[String]) => array ++ Array(""))
+
+  override def fit(dataset: Dataset[_]): H2OMOJOModel = {
+    val inputCol = getInputCol
+    val ds = dataset
+      .withColumn(inputCol, addValue(col(inputCol)))
+      .withColumn(inputCol, explode(col(inputCol)))
+      .select(inputCol)
+
+    val h2oContext = H2OContext.ensure(
+      "H2OContext needs to be created in order to train the model. Please create one as H2OContext.getOrCreate().")
+    val train = h2oContext.asH2OFrame(ds.toDF())
+    train.convertColumnsToStrings(Array(0))
+    val params = getH2OAlgorithmParams(train) ++
+      Map("training_frame" -> train.frameId, "model_id" -> convertModelIdToKey(getModelId()))
+    val modelId = trainAndGetDestinationKey(s"/3/ModelBuilders/word2vec", params)
+    deleteRegisteredH2OFrames()
+    val settings = H2OMOJOSettings(predictionCol = getOutputCol)
+    H2OModel(modelId)
+      .toMOJOModel(Identifiable.randomUID(classOf[H2OWord2Vec].getSimpleName), settings)
+  }
+
+  def setInputCol(value: String): this.type = set(inputCol, value)
+
+  def setOutputCol(value: String): this.type = set(outputCol, value)
+
+  @DeveloperApi
+  override def transformSchema(schema: StructType): StructType = schema
```

Comment:
Shouldn't be some parameter checks performed here? InputCol is set and in the dataset, column types, etc.

Reply:
Yup, completely missed this part. Thanks for checking. I will also add a test